tests/models/msclap/test_modeling_msclap.py

```diff
@@ -0,0 +1,709 @@
+                key: value for key, value in loaded_model_state_dict.items() if key not in non_persistent_buffers
+            }
+
+            self.assertEqual(set(model_state_dict.keys()), set(loaded_model_state_dict.keys()))
+
+            model_buffers = list(model.buffers())
+            for non_persistent_buffer in non_persistent_buffers.values():
+                found_buffer = False
+                for i, model_buffer in enumerate(model_buffers):
+                    if torch.equal(non_persistent_buffer, model_buffer):
+                        found_buffer = True
+                        break
+
+                self.assertTrue(found_buffer)
+                model_buffers.pop(i)
+
+            models_equal = True
+            for layer_name, p1 in model_state_dict.items():
+                p2 = loaded_model_state_dict[layer_name]
+                if p1.data.ne(p2.data).sum() > 0:
+                    models_equal = False
+
+            self.assertTrue(models_equal)
+
+    def test_load_audio_text_config(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+
+        # Save MSClapConfig and check if we can load MSClapAudioConfig from it
+        with tempfile.TemporaryDirectory() as tmp_dir_name:
+            config.audio_config.save_pretrained(tmp_dir_name)
+            audio_config = MSClapAudioConfig.from_pretrained(tmp_dir_name)
+            self.assertDictEqual(config.audio_config.to_dict(), audio_config.to_dict())
+
+        # Save MSClapConfig and check if we can load MSClapTextConfig from it
+        with tempfile.TemporaryDirectory() as tmp_dir_name:
+            config.text_config.save_pretrained(tmp_dir_name)
+            text_config = MSClapTextConfig.from_pretrained(tmp_dir_name)
+            self.assertDictEqual(config.text_config.to_dict(), text_config.to_dict())
+
+    @slow
+    def test_model_from_pretrained(self):
+        model_name = "kamilakesbi/ms_clap"
+        model = MSClapModel.from_pretrained(model_name)
+        self.assertIsNotNone(model)
+
+
+@slow
+@require_torch
+class MSClapModelIntegrationTest(unittest.TestCase):
+    def test_integration_unfused(self):
+        EXPECTED_MEANS_UNFUSED = 6.15e-05
+
+        librispeech_dummy = load_dataset(
+            "hf-internal-testing/librispeech_asr_dummy", "clean", split="validation", trust_remote_code=True
+        )
+        audio_sample = librispeech_dummy[-1]
+
+        model_id = "kamilakesbi/ms_clap"
+
+        model = MSClapModel.from_pretrained(model_id).to(torch_device)
+        processor = MSClapProcessor.from_pretrained(model_id)
+
+        # for padding in self.paddings:
+        inputs = processor(audios=audio_sample["audio"]["array"], return_tensors="pt").to(torch_device)
+
+        audio_embed = model.get_audio_features(**inputs)
+        expected_mean = EXPECTED_MEANS_UNFUSED
+
+        self.assertTrue(torch.allclose(audio_embed.cpu().mean(), torch.tensor([expected_mean]), atol=1e-3, rtol=1e-3))
```

Comment:
The tolerance threshold is a bit high given the value of the expected mean, do you think you could lower the threshold?